packages/fuels-signers/src/wallet.rs

```diff
@@ -899,4 +904,138 @@ mod tests {
+                ),
+                other => other.clone(),
+            })
+            .collect();
+
+        let comparison_results: Vec<bool> = inputs
+            .iter()
+            .map(|input| {
+                let found_index = expected_inputs
+                    .iter()
+                    .position(|expected| expected == input);
+                if let Some(index) = found_index {
+                    expected_inputs.remove(index);
+                    true
+                } else {
+                    false
+                }
+            })
+            .collect();
+
+        if !expected_inputs.is_empty() {
+            return false;
+        }
+
+        return comparison_results.iter().all(|&r| r);
+    }
+
+    #[tokio::test]
+    async fn add_fee_coins_empty_transaction() -> Result<(), Error> {
+        let wallet_config = add_fee_coins_wallet_config(1);
+        let wallet = launch_custom_provider_and_get_wallets(wallet_config, None)
+            .await
+            .pop()
+            .unwrap();
+        let mut tx = Wallet::build_transfer_tx(&[], &[], TxParameters::default());
+        wallet.add_fee_coins(&mut tx, 0, 0).await?;
+
+        let zero_utxo_id = UtxoId::new(Bytes32::zeroed(), 0);
+        let mut expected_inputs = vec![Input::coin_signed(
+            zero_utxo_id,
+            wallet.address().into(),
+            20,
+            BASE_ASSET_ID,
+            TxPointer::default(),
+            0,
+            0,
+        )];
+        let expected_outputs = vec![Output::change(wallet.address().into(), 0, BASE_ASSET_ID)];
+
+        assert!(compare_inputs(tx.inputs(), &mut expected_inputs));
+        assert_eq!(tx.outputs(), expected_outputs);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn add_fee_coins_to_transfer_with_base_asset() -> Result<(), Error> {
+        let wallet_config = add_fee_coins_wallet_config(1);
+        let wallet = launch_custom_provider_and_get_wallets(wallet_config, None)
+            .await
+            .pop()
+            .unwrap();
+
+        let base_amount = 30;
+        let inputs = wallet
+            .get_asset_inputs_for_amount(BASE_ASSET_ID, base_amount, 0)
+            .await?;
+
+        let outputs = wallet.get_asset_outputs_for_amount(
+            &Address::zeroed().into(),
+            BASE_ASSET_ID,
+            base_amount,
+        );
+
+        let mut tx = Wallet::build_transfer_tx(&inputs, &outputs, TxParameters::default());
+        wallet.add_fee_coins(&mut tx, base_amount, 0).await?;
+
+        let zero_utxo_id = UtxoId::new(Bytes32::zeroed(), 0);
+        let mut expected_inputs: Vec<Input> = (0..3)
+            .map(|_| {
+                Input::coin_signed(
+                    zero_utxo_id,
+                    wallet.address().into(),
+                    20,
+                    BASE_ASSET_ID,
+                    TxPointer::default(),
+                    0,
+                    0,
+                )
+            })
+            .collect();
```

Comment:
Another way of writing it which doesn't require ignoring the counter:
```rust
 let mut expected_inputs = repeat(Input::coin_signed(
            zero_utxo_id,
            wallet.address().into(),
            20,
            BASE_ASSET_ID,
            TxPointer::default(),
            0,
            0,
        ))
        .take(3)
        .collect::<Vec<_>>();
```
where `repeat` is `std::iter::repeat`. It generates an infinite iterator that repeats the given element from which you only take 3 elements.